src/Authentication/Traits/HasAccessTokens.php

```diff
@@ -165,4 +171,64 @@ public function setAccessToken(?AccessToken $accessToken): self
 
         return $this;
     }
+
+    /**
+     * Checks if the provided Access Token has expired.
+     *
+     * @return bool|null Returns true if Access Token has expired, false if not, and null if the expire field is null
+     */
+    public function hasAccessTokenExpired(?AccessToken $accessToken): bool|null
```

Comment:
Can we return here only `true`/`false` - this seems more appropriate.

We can introduce another method, like `canAccessTokenExpire()` to determine whether the token has an expiration date.